File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Math/Add_Binary/add_binary.js
// Add Binary - JavaScript implementation
// Time Complexity: O(max(m, n)) where m and n are the lengths of a and b
// Space Complexity: O(max(m, n)) for the result string

function addBinary(a, b) {
  let i = a.length - 1;
  let j = b.length - 1;
  let carry = 0;
  const result = [];

  // Process from right to left (least significant to most significant)
  while (i >= 0 || j >= 0 || carry > 0) {
    const digitA = i >= 0 ? parseInt(a[i]) : 0;
    const digitB = j >= 0 ? parseInt(b[j]) : 0;

    const sum = digitA + digitB + carry;
    result.push((sum % 2).toString());
    carry = Math.floor(sum / 2);

    i--;
    j--;
  }

  // Reverse to get the correct order (most significant first)
  return result.reverse().join("");
}

module.exports = { addBinary };

File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Math/Add_Binary/add_binary.cs
// Add Binary - C# implementation
// Time Complexity: O(max(m, n)) where m and n are the lengths of a and b
// Space Complexity: O(max(m, n)) for the result string

public class Solution
{
    public string AddBinary(string a, string b)
    {
        int i = a.Length - 1;
        int j = b.Length - 1;
        int carry = 0;
        var result = new System.Collections.Generic.List<char>();

        // Process from right to left (least significant to most significant)
        while (i >= 0 || j >= 0 || carry > 0)
        {
            int digitA = i >= 0 ? a[i] - '0' : 0;
            int digitB = j >= 0 ? b[j] - '0' : 0;

            int sum = digitA + digitB + carry;
            result.Add((char)((sum % 2) + '0'));
            carry = sum / 2;

            i--;
            j--;
        }

        // Reverse to get the correct order (most significant first)
        result.Reverse();
        return new string(result.ToArray());
    }
}

